### DISCUSSION TITLE
关于配置的一些问题

### DISCUSSION THREAD

**Discussion by ggtony233** [XTLS/Xray-core#2947] [Q&A]
完完全全的小白一枚，由于腾讯买的新人福利轻量云过期了，本身也有翻墙需求，遂购买了一年阿里云的新加坡vps用于穿透家庭内网和科学冲浪，之前在网上搜到一篇和wireguard有关的文章,[使用v2ray对wireguard流量进行加密的udp overtcp方案](https://github.com/scriptwang/blog/blob/master/docs/blog/2021-12-26_WireGuard%2BV2ray%E6%89%93%E9%80%A0%E5%8F%B2%E4%B8%8A%E6%9C%80%E5%AE%89%E5%85%A8%E9%9A%90%E5%8C%BF%E4%B8%94%E9%AB%98%E9%80%9F%E7%9A%84%E5%86%85%E7%BD%91%E7%A9%BF%E9%80%8F%E9%9A%A7%E9%81%93.md)  
狠狠的心动，但碍于本人可怜的知识储备，只能求助各位大佬了，如果要用类似于上面的方案对wireguard 的udp流量通过xray套一层tcp保证连接成功和原本的魔法上网功能，通过VLESS-XTLS-uTLS-REALITY的方式去配置的话，我应当如何设置出站与进站呢？我想到下面几个问题:
- 如何设置出站来确保vps 的wireguard服务端能够正常的从xray接收客户端发来的流量？任意门吗？
- 客户端的设置是否和上面那篇文章一样在入站设置中监听本地wireguard端口，然后在wireguard中将端点ip设置为localhost？
==============
还有几个疑问
科学上网能否和上面说的wireguard加密同时实现在一个xray进程中？可以的话客户端的进站和服务端出站应当如何设置？  
如果要让客户端的请求流量过nginx的443端口在到达用VLESS-XTLS-uTLS-REALITY的配置方式启动的xray中，nginx部分该如何去写呢？

**Comment by chika0801** [comment]
> 如果要用类似于上面的方案对wireguard 的udp流量通过xray套一层tcp保证连接成功和原本的魔法上网功能

建议放弃上面文章中的方案，你在Xray社区来问，就用这边现在用的人多的那些组合形式。

**Reply by chika0801** [reply]
https://github.com/chika0801/Xray-examples

常用的一些组合形式

**Reply by ggtony233** [reply]
主要是自己也有内网穿透需求啊
